src/replay.ts

```diff
@@ -299,6 +259,160 @@ function validateReplayNormalizedOptions(fromDate: Date, normalizers: MapperFact
   }
 }
 
+async function* createMessageStream(cachedSlicePath: string, withDisconnects: boolean) {
+  let replayError
+  // response is a path to file on disk let' read it as stream
+  const linesStream = createReadStream(cachedSlicePath, { highWaterMark: 128 * 1024 })
+    // unzip it
+    .pipe(zlib.createGunzip({ chunkSize: 128 * 1024 }))
+    .on('error', (err) => {
+      debug('gunzip error %o', err)
+      replayError = err
+    })
+    // and split by new line
+    .pipe(new BinarySplitStream())
+
+  // date is always formatted to have lendth of 28 so we can skip looking for first space in line and use it
+  // as hardcoded value
+  const DATE_MESSAGE_SPLIT_INDEX = 28
+
+  // helper flag that helps us not yielding two subsequent undefined/disconnect messages
+  let lastMessageWasUndefined = false
+  for await (const line of linesStream) {
+    if (replayError) throw replayError
+    const bufferLine = line as Buffer
+
+    if (bufferLine.length > 0) {
+      lastMessageWasUndefined = false
+      const localTimestampBuffer = bufferLine.slice(0, DATE_MESSAGE_SPLIT_INDEX)
+      const messageBuffer = bufferLine.slice(DATE_MESSAGE_SPLIT_INDEX + 1)
+      // as any due to https://github.com/Microsoft/TypeScript/issues/24929
+      yield {
+        localTimestamp: localTimestampBuffer,
+        message: messageBuffer
+      } as any
+      // ignore empty lines unless withDisconnects is set to true
+      // do not yield subsequent undefined messages
+    } else if (withDisconnects === true && lastMessageWasUndefined === false) {
+      lastMessageWasUndefined = true
+      yield undefined as any
+    }
+  }
+}
+
```

Comment:
Message decoding logic is extracted, because when merging multiple message streams, I depend on `.localTimestamp` as raw Buffer.